scripts/policy/protocols/ssl/decryption.zeek

```diff
@@ -109,3 +105,10 @@ event ssl_change_cipher_spec(c: connection, is_client: bool)
 			set_secret(c, secrets[c$ssl$client_random]);
 		}
 	}
+
+# Do not disable analyzers after detection - otherwise we will not receive
+# encrypted packets.
+hook Analyzer::disabling_analyzer(c: connection, atype: AllAnalyzers::Tag, aid: count)
+    {
+    break;
```

Comment:
This makes sense, but it's missing one crucial part: `Analyzer::disabling_analyzer()` is invoked for every analyzer on every connection that's about to be disabled. You'd want to prevent disabling only the right SSL analyzer.

So this should be something like:
```
hook Analyzer::disabling_analyzer(c: connection, atype: AllAnalyzers::Tag, aid: count)
    {
    # Don't care about non-SSL analyzers.
    if ( atype != Analyzer::ANALYZER_SSL )
        return;

    # Only prevent disabling SSL analyzers of this connection.
    if ( c?$ssl && c$ssl?$analyzer_id && c$ssl$analyzer_id == aid )
        break;
    }
```

Reply:
I'm wondering a bit if we should keep track of SSL analyzer violations though. If they happen, allow the disabling. Otherwise, with this change, SSL analyzer violations won't disable the analyzer anymore.